beesdoo_product/tests/test_search_main_seller_product_code.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2021 Coop IT Easy SCRL fs
+#   Rémy Taymans <remy@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+
+from odoo.tests import TransactionCase
+
+
+class TestSearchMainSellerProductCode(TransactionCase):
+    def setUp(self):
+        super().setUp()
+        self.product1 = self.env.ref("product.product_delivery_01")
+        self.product2 = self.env.ref("product.product_delivery_02")
+
+        for index, sup in enumerate(self.product1.seller_ids):
+            sup.product_code = "AAA%d" % index
+        for index, sup in enumerate(self.product2.seller_ids):
+            sup.product_code = "BBB%d" % index
+
+        # Try not to use _get_main_supplier_info as it is a private
+        # function. I use main_seller_id instead, but main_seller_id
+        # does not contain the supplierinfo, just the res.partner.
+        def get_main_supplier_info(product):
+            for sup in product.seller_ids:
+                if sup.name == product.main_seller_id:
+                    return sup
+            return self.env["product.supplierinfo"]
+
+        self.product1_main_supplierinfo = get_main_supplier_info(self.product1)
+        self.product2_main_supplierinfo = get_main_supplier_info(self.product2)
+
+    def test_search_product_by_main_seller_product_code(self):
+        """
+        Test searching a product giving the product_code of it's
+        main_seller_id
+        """
+        products = self.env["product.product"].search(
```

Comment:
I think this search does not match the GUI search. Here you can search on computed non-stored fields but not in the GUI. But I may be wrong. And I like tests.

Reply:
If your goal was rather to test your compute function, I'd name the test function differently.

Reply:
Ok, will change the name. Not sure how to test the GUI search. I was thinking that is use simply the `search` function.

Reply:
In fact, my function was really to test that the `search` function works well on the related (previously computed) field. May be another fonction can be written to test the compute function.